Downloader.m

```diff
@@ -118,4 +126,19 @@ - (void)stopDownload
   return _params.errorCallback(error);
 }
 
+
+- (void)URLSessionDidFinishEventsForBackgroundURLSession:(NSURLSession *)session
+{
+    NSLog(@"URLSessionDidFinishEventsForBackgroundURLSession");
+
+    AppDelegate *appDelegate = (AppDelegate *)[[UIApplication sharedApplication] delegate];
+
+    
+    if (appDelegate.sessionCompletionHandler) {
```

Comment:
exactly, you have up to 30 seconds to call the completionHandler, otherwise the app crashes due to this assertion:
```
        <BKProcessAssertion: 0x157524050> id: 6044-13B80FC9-6760-46DE-A46A-FAFC5B88540D name: com.apple.nsurlsessiond.handlesession 8AA63A90-6574-44F4-9DAB-DD4ADFD9893D process: <BKNewProcess: 0x157525520; com.commercialtribe.mobile.dev; pid: 7112; hostpid: -1> permittedBackgroundDuration: 30.000000 reason: backgroundDownload owner pid:6044 preventSuspend  preventThrottleDownUI  preventIdleSleep  preventSuspendOnSleep
````

So I store the handler in our AppDelegate here https://github.com/k7k0/mobile/commit/4b3046148bd32a010b7087d8eeb7829a5916d834#diff-05fad48601d118a299c7de5409225ccfR64

To keep the fs library organized we should create a `setCompletionHandler` there and invoque it from our AppDelegate. I couldn't found how tough (finding the right instance of `Downlader`), one option would be storing a class-level static variable (as we dont have N concurrent BG processes)